doc(website): Add Docs for AWS `table_options` (#10804)
cloudquery/cloudquery: website/pages/docs/plugins/sources/aws/configuration.md

  When set to `true` plugin will sync data from APIs that incur a fee. Currently only `aws_costexplorer*` tables require this flag to be set to `true`.

- **experimental** `table_options` (map) (default: not used)

  This is an experimental feature that enables users to override the default options for specific tables. The root of the object takes a table name, and the next level takes an API method name. The final level is the actual input object as defined by the API. 
  The format of the `table_options` object is as follows:
  ```yaml
  table_options:
    <table_name>:
      <api_method_name>:
        - <input_object>
  ```

>   For the `<input_object>` it is able to take a list of objects which cloudquery will iterate through to make multiple API calls. This is useful for APIs like CloudTrail's `LookupEvents` that only supports a single event type per call. for example:

Reviewer: ```suggestion
  A list of `<input_object>` objects should be provided. CloudQuery will iterate through these to make multiple API calls. This is useful for APIs like CloudTrail's `LookupEvents` that only supports a single event type per call. For example:
```

>   ```yaml
>         table_options:
>       aws_cloudtrail_events:
>         lookup_events:
>           - start_time: 2023-05-01T20:20:52Z

Reviewer: ```suggestion
  ```yaml
    table_options:
      aws_cloudtrail_events:
        lookup_events:
          - start_time: 2023-05-01T20:20:52Z
```

            end_time:   2023-05-03T20:20:52Z
            lookup_attributes:
              - attribute_key: EventName
                attribute_value: RunInstances
          - start_time: 2023-05-01T20:20:52Z
            end_time:   2023-05-03T20:20:52Z
            lookup_attributes:
              - attribute_key: EventName
                attribute_value: StartInstances
          - start_time: 2023-05-01T20:20:52Z
            end_time:   2023-05-03T20:20:52Z
            lookup_attributes:
              - attribute_key: EventName
                attribute_value: StopInstances
  ```

>   The naming for all of the fields is the same as the AWS API but in snake case. For example `EndTime` is represented as `end_time`. As of `VERSION_SOURCE_AWS` the following tables and APIs are supported:

Reviewer: ```suggestion
  The naming for all of the fields is the same as the AWS API but in snake case. For example `EndTime` is represented as `end_time`. As of `v17.3.0` the following tables and APIs are supported:
```

`VERSION_SOURCE_AWS` only gets replaced in triple-tick code blocks
